test/e2e/funcs/feature.go

```diff
@@ -219,7 +219,11 @@ func ResourcesHaveConditionWithin(d time.Duration, dir, pattern string, cds ...x
 			_ = fieldpath.Pave(u.Object).GetValueInto("status", &s)
 
 			for _, want := range cds {
-				if !s.GetCondition(want.Type).Equal(want) {
+				got := s.GetCondition(want.Type)
```

Comment:
Same as https://github.com/crossplane/crossplane/pull/4584#discussion_r1325368954

Reply:
Updated. Ptal. Now it's unconditional, documented and the dev gets an error if a message is set in the comparison conditions.

Reply:
This might break some existing e2e tests, did you check that?

Reply:
Actually, you are right. It could break something :) Checking.

Reply:
Double checked. We should be fine.